includes/amp-helper-functions.php

```diff
@@ -1450,6 +1450,16 @@ function amp_is_dev_mode() {
 			( is_admin_bar_showing() && is_user_logged_in() )
 			||
 			is_customize_preview()
+			||
+			(
+				! is_ssl()
+				&&
+				(
+					( function_exists( 'wp_get_environment_type' ) && 'local' === wp_get_environment_type() )
+					||
+					'localhost' === wp_parse_url( home_url(), PHP_URL_HOST )
+				)
```

Comment:
I wonder if this would make sense:

```suggestion
				'localhost' === wp_parse_url( home_url(), PHP_URL_HOST )
```

The reason being that the URLs in question only work if on `localhost` in the first place, so therefore does it even matter what `wp_get_environment_type()` returns?

Reply:
Or alternatively:

```suggestion
				(
					function_exists( 'wp_get_environment_type' )
					&&
					'local' === wp_get_environment_type()
					&&
					'localhost' === wp_parse_url( home_url(), PHP_URL_HOST )
				)
```

This would make `wp_get_environment_type()` a guard on whether this is opted-into.

Reply:
Naturally, the outer parentheses could be removed as well in this case:

```suggestion
				function_exists( 'wp_get_environment_type' )
				&&
				'local' === wp_get_environment_type()
				&&
				'localhost' === wp_parse_url( home_url(), PHP_URL_HOST )
```

Reply:
> The reason being that the URLs in question only work if on localhost in the first place, so therefore does it even matter what wp_get_environment_type() returns?

There can be a use case when the user can try to emulate a production-like environment by setting up `WP_ENVIRONMENT_TYPE` env. So the latter option makes sense to me.